src/punica.c

```diff
@@ -90,6 +91,51 @@ static void init_signals(void)
     }
 }
 
+static int plugins_load(basic_plugin_manager_t *plugin_manager,
+                        plugins_settings_t *plugins_settings)
+{
+    linked_list_entry_t *entry;
+    plugin_settings_t *plugin;
+    int load_is_successful;
+
+    for (entry = plugins_settings->plugins_list->head;
+         entry != NULL; entry = entry->next)
+    {
+        plugin = entry->data;
+        load_is_successful = basic_plugin_manager_load_plugin(plugin_manager,
+                                                              plugin->path,
+                                                              plugin->name);
+
+        if (load_is_successful)
+        {
+            log_message(LOG_LEVEL_INFO,
+                        "[PLUGINS] Successfuly loaded plugin: \"%s\"\n",
+                        plugin->name);
+        }
+        else
+        {
+            log_message(LOG_LEVEL_ERROR,
+                        "[PLUGINS] Failed to load plugin: \"%s\"\n", plugin->name);
+        }
+    }
+
+    return 0;
+}
+
+static void plugins_unload(plugins_settings_t *plugins_settings)
+{
+    linked_list_entry_t *entry;
+    plugin_settings_t *plugin;
+
+    for (entry = plugins_settings->plugins_list->head;
+         entry != NULL; entry = entry->next)
+    {
+        plugin = entry->data;
+        free(plugin);
+    }
+    free(plugins_settings->plugins_list);
```

Comment:
Freeing in such a way leaves linked list entries unfreed. Should use `linked_list_delete()`